packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -588,6 +590,20 @@ class SliderThemeData with Diagnosticable {
   /// If specified, overrides the default value of [Slider.allowedInteraction].
   final SliderInteraction? allowedInteraction;
 
+  /// The size of the [BarSliderThumbShape] thumb.
+  ///
+  /// If [SliderThemeData.thumbShape] is [BarSliderThumbShape], this property is used to
+  /// set the size of the thumb. Otherwise, the default thumb size is 4 pixels for the
+  /// width and 44 pixels for the height.
+  final MaterialStateProperty<Size?>? barThumbSize;
+
+  /// The size of the gap between the active and inactive tracks of the [GappedSliderTrackShape].
```

Comment:
Is this only relevant for the M3 slider? Or both? We should check that the docs clearly state when/how a parameter/behavior/etc applies for M2 versus M3

Reply:
All the new shapes are compatible with M2/M3.  Great suggestion

I added this documentation 
```dart
  /// The Slider defaults to [GappedSliderTrackShape] when the track shape is
  /// not specified, and the [trackGapSize] can be used to adjust the gap size.
  ///
  /// If the [ThemeData.useMaterial3] is false, then the Slider track shape
  /// defaults to [RoundedRectSliderTrackShape] and the [trackGapSize] is ignored.
  /// In this case, set the track shape to [GappedSliderTrackShape] to use the
  /// [trackGapSize].
  final double? trackGapSize;
```